### PR TITLE
Default market strategy: least expensive providers; choose random offer among offers with the highest score

### PR DESCRIPTION
**Pull request by filipgolem**

### CODE REVIEW

**Comment by shadeofblue** [comment]
I wouldn't merge this into b0.2

**Comment by filipgolem** [comment]
> I wouldn't merge this into b0.2

I was not sure which branch to choose (b0.2 or master).

File: yajsapi/runner/index.ts [hunk 1]
```diff
@@ -608,20 +608,23 @@ export class Engine {
...
           let _offer_list = Object.entries(offer_buffer);
           let _sample =
-            _offer_list[
-              Math.floor(Math.random() * Object.keys(offer_buffer).length)
-            ];
+            _offer_list
+              .map(x => { return { obj: x, rnd: Math.random() }; })
+              .sort((a, b) => a.rnd - b.rnd)
+              .map(x => x.obj)
+              .reduce((item, acc) => item[1].score > acc[1].score ? item : acc);
```
Comment (mdtanrikulu) [inline yajsapi/runner/index.ts:620]: order of props should be accumulator, item

Comment (filipgolem) [reply yajsapi/runner/index.ts]: Fixed

**Review by mdtanrikulu — APPROVED** [review]

**Pull request merged**